lib/metrics.py

```diff
@@ -348,3 +352,79 @@ def _get_db_conn(self):
                 psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT
             )
         return self.db
+
+
+class PaidAppsMetricsEmitterThread(BaseMetricsEmitterThread):
+    @property
+    def _select_stats_to_emit(self):
+        selected_stats = []
+        if buildpackutil.i_am_primary_instance():
+            selected_stats = [
+                self._inject_database_stats,
+                self._inject_storage_stats,
+                self._inject_health,
+            ]
+        selected_stats.append(self._inject_m2ee_stats)
+        return selected_stats
+
+    def _gather_metrics(self):
+        stats = {}
+        try:
+            for inject_method in self._select_stats_to_emit:
+                stats = inject_method(stats)
+        except psycopg2.OperationalError as exc:
+            logger.exception("METRICS: error while gathering metrics")
+            stats = {
+                "health": {
+                    "health": 0,
+                    "diagnosis": "Database error: {}".format(str(exc)),
+                }
+            }
+        except Exception:
+            logger.exception("METRICS: error while gathering metrics")
+            stats = {
+                "health": {
+                    "health": 4,
+                    "diagnosis": "Unable to retrieve metrics",
+                }
+            }
+        finally:
+            return stats
+
+
+class FreeAppsMetricsEmitterThread(BaseMetricsEmitterThread):
+    def _get_munin_stats(self):
+        m2ee_stats, _ = munin.get_stats_from_runtime(
+            self.m2ee.client, self.m2ee.config
+        )
+        return m2ee_stats
+
+    def _inject_user_session_metrics(self, stats):
+        session_metrics = {}
+        try:
+            m2ee_stats = self._get_munin_stats()
+            if "sessions" in m2ee_stats.keys():
```

Comment:
You don't need to call the `keys()` function here. You can just use `if x in y`:
```
>>> a  = {'foo': 123, 'bar': 456}
>>> 'foo' in a
True
>>> 'baz' in a
False
>>>
```

```suggestion
            if "sessions" in m2ee_stats:
```